tests/Hateoas/Tests/Fixtures/Gh236Bar.php

```diff
@@ -13,6 +13,7 @@ class Gh236Bar
 
     /**
      * @Serializer\Expose()
+     * @Serializer\SkipWhenEmpty()
```

Comment:
Without this option the output of the [test XmlHalSerializerTest](https://github.com/willdurand/Hateoas/blob/7e1a2339c66da73f2b5d88f902bf2c9ba5c4ee02/tests/Hateoas/Tests/Serializer/XmlHalSerializerTest.php#L61) will be:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<collection>
  <resource rel="items">
    <a>
      <xxx><![CDATA[yyy]]></xxx>
      <inner/> <<<<--- this should not be here
    </a>
    <resource rel="b_embed">
      <xxx><![CDATA[zzz]]></xxx>
      <inner/> <<<<--- this should not be here 
    </resource>
  </resource>
</collection>
```

the properties  from the object representing `<inner/>` are excluded by the max depth exclusion strategy, so there is no point in having the empty node there